src/backend_model.cc

```diff
@@ -330,6 +330,22 @@ TritonModel::UpdateModelConfig(
   auto outputs_config = config.mutable_output();
   *outputs_config = updated_config.output();
 
+  if (!config.scheduling_choice_case() ) {
```

Comment:
Do we need an else if clause to fail the auto-complete if both the original and updated config has a scheduling setting?

Reply:
I don't believe we should fail here. It's up to each backend how they want to autocomplete the scheduling choice. The else-if in this case being the user has provided a scheduling choice in their model config and autocomplete believing it should be something different. 

An example might be, the user sets the sequence batching in their model config but autocomplete wants to set it to dynamic batching. Guan's point above is that we don't know if  autocomplete checked this scheduling choice before it sets it. Thus, only set the scheduling choice if none has been set and do nothing otherwise (prioritizing the user's config). 

Let me know if this was incoherent and needs more explanation.

Reply:
I was wondering may be we should remove the condition `!config.scheduling_choice_case()` altogether. If I'm understanding correctly, it looks like we will be silently not setting what the backend has provided to us as an auto-completed model configuration for scheduling section.

Similar to how we don't do any error checking for inputs/outputs, I think we should trust that the backend developer has properly auto-filled the scheduling section and we will use whatever setting they have provided as the final configuration. All the conflict detection between the original and the auto-completed model configuration should be handled in the backend side (similar to inputs and outputs).

Reply:
This [idea was rejected](https://github.com/triton-inference-server/core/pull/79#discussion_r881008560) by @GuanLuo above with the following reason: 
> the danger comes from when backend implementer (/ user) tries to have auto-complete support that always set dynamic batching when possible, without realizing that the model may have been configured as a sequence model.

Reply:
I find it a bit strange that here we are neither returning an error nor setting what the backend has specified. I think maybe we can wait until Guan is back to make the final call.

Reply:
I think we can return error here if the backend tries to overwrite the scheduling choice

Reply:
I may need a bit of clarification here. From this thread I understand the following:
- We will only set the scheduling choice if none is already set
- If the scheduling choice in `config` is different from the scheduling choice in `updated_config`, we will return an Error, presumably INTERNAL, with an appropriate error message.
- If the scheduling choice in `config` is the same as the scheduling choice in `updated_config` then skip this update.

Is this the correct understanding?

Reply:
Yes, that is my understanding as well, although I would use `INVALID_ARGS` instead of `INTERNAL` because the error returned in this case is intentional.

CC @tanmayv25 to see if below make sense: My thought process is that comparing to I/O where the backend will know better and can provide "correct" config by examining the model file, scheduling choice is different in the sense that if user specify it in model config, then the backend shouldn't overwrite it as the user "know better" on how the model should be served

Reply:
I agree with current changes. We must through an error if the backend developer sets a scheduling_choice incorrectly.
The autocomplete logic in  backend must be mindful of handling whatever scheduling choice is specified in config.pbtxt and not try to change it. 
We have those checks in TF and ORT backends before we try adding dynamic_batching scheduling choice.

Reply:
I think it should still be INTERNAL because the backend developer has not done a good job of writing a robust auto-complete logic.